### ISSUE TITLE
some questions about the paper

### ISSUE THREAD

**Issue by songyonger** [plumerai/rethinking-bnn-optimization#26]
Thanks for your work! Now i have some questions about the  paper.
①"For example, when using SGD and Glorot initialization [11] a learning rate of 1 performs much better than 0.01; but when we multiply the initialized weights by 0.01 before starting training, we obtain the same improvement in performance." But according to the theorem 1: binary weight is invariant if the initial weight and learning rate are scaled a scalar simultaneously , so i want to know why multiply initialized weights by 0.01  will achieve the same improvement as using a learning rate of 1 ？ and how does decreasing the learning rate build up inertia?
②how does the moving average  filtering out short-lived  signals ？ 
③in algorithm 2, why must have the condition "sign(mi)=sign(wi)" then the weight flip? 
@jamescook106 @lgeiger @koenhelwegen

**Comment by koenhelwegen** [comment]
Thanks for the interest in our work!

> why multiply initialized weights by 0.01 will achieve the same improvement as using a learning rate of 1?

The key observation is that what matters is the magnitude of the learning rate with respect to the distribution of the weights. To understand why, I recommend you have a look at the proof of the theorem.

> and how does decreasing the learning rate build up inertia?

See again the proof: if we decrease the learning rate, you effectively need a larger signal to flip the sign of a given latent weight.

> how does the moving average filtering out short-lived signals?

Suppose there is a strong signal during a single step that disappears afterwards. Without the moving average (or some other 'state'), any updating rule would take such a large signal and flip the weight. However, this is probably not optimal - hence we use the moving average.

> why must have the condition "sign(mi)=sign(wi)" then the weight flip?

The gradient is with respect to the loss; so the direction of the gradient indicates the direction in which the loss increases. This is the opposite of what we want, which is why in normal networks we apply gradient descent (instead of ascent). Similarly, in the case of BNNs a positive gradient on a positive weight indicates we should flip the weight.

**Comment by songyonger** [comment]
Thank you very much! will close the issue.
